aten/src/ATen/native/TensorShape.cpp

```diff
@@ -49,7 +49,73 @@ Tensor & cat_out(Tensor & result, TensorList tensors, int64_t dim) {
   return at::_th_cat_out(result, tensors, dim);
 }
 
+static bool sizes_match_except(IntList s1, IntList s2, int64_t dim_except) {
+  if (s1.size() != s2.size()) {
+    return false;
+  }
+  for (int64_t i = 0; i < s1.size(); ++i) {
+    if (i != dim_except && s1[i] != s2[i]) {
```

Comment:
I don't think this will work with wrapped dimensions (negative dimensions count from the end).  Can you add a test?  Probably need to call maybe_wrap_dim earlier.